src/components/container/OverviewTable/OverviewTable.tsx

```diff
@@ -0,0 +1,146 @@
+import { LayoutContext, LoadingIndicator } from '../../presentational';
+import './OverviewTable.css';
+import { add, startOfToday } from 'date-fns';
+import { createPreviewDataProvider } from './OverviewTable.previewData';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import language from '../../../helpers/language';
+
+export interface OverviewTableProps {
+    preview?: boolean;
+    primaryDataType: PrimaryOverviewDataType;
+    secondaryDataTypes: SecondaryOverviewDataType[];
+    daysToCompute?: number;
+    primaryHeaderBackgroundColor?: ColorDefinition;
+    primaryHeaderTextColor?: ColorDefinition;
+    goodValueBackgroundColor?: ColorDefinition;
+    goodValueTextColor?: ColorDefinition;
+    goodValueIndicator?: IconDefinition;
+    notGoodValueBackgroundColor?: ColorDefinition;
+    notGoodValueTextColor?: ColorDefinition;
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+/**
+ * The overview table component can be used to compare multiple daily survey and daily data values in a tabular format.
+ *
+ * The primary data type is first grouped by value threshold.  The secondary data types are then grouped by the same
+ * days in which the primary values met each threshold.  This is intended to illustrate what the values from each of
+ * the secondary data types looked like on the days when the values from the primary data type were within a given
+ * threshold (i.e. What did my step count look like when my mood rating was high?).
+ */
+export default function OverviewTable(props: OverviewTableProps) {
+    const layoutContext = useContext(LayoutContext);
+
+    const [primaryData, setPrimaryData] = useState<OverviewData<PrimaryOverviewDataType>>();
+    const [secondaryData, setSecondaryData] = useState<OverviewData<SecondaryOverviewDataType>[]>();
+
+    const primaryHeaderBackgroundColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderBackgroundColor ?? { lightMode: '#e2f1ff', darkMode: '#000' });
+    const primaryHeaderTextColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderTextColor ?? { lightMode: '#000', darkMode: '#8bcbff' });
+
+    const goodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.goodValueBackgroundColor ?? { lightMode: '#c8ffc2', darkMode: '#000' });
+    const goodValueTextColor = resolveColor(layoutContext.colorScheme, props.goodValueTextColor ?? { lightMode: '#000', darkMode: '#96ff85' });
+
+    const notGoodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.notGoodValueBackgroundColor ?? { lightMode: '#ffedc2', darkMode: '#000' });
+    const notGoodValueTextColor = resolveColor(layoutContext.colorScheme, props.notGoodValueTextColor ?? { lightMode: '#000', darkMode: '#ffc847' });
+
+    const daysToCompute = props.daysToCompute ?? 28;
```

Comment:
Is 28 used bc of days in month? If yes, can getDaysInMonth be used?

Reply:
The 28 day default is because the original use case for this component was to compute data from the past 4 weeks.  We can pick a different default, but I think it should be a fixed number for consistency rather than something variable like the number returned by `getDaysInMonth`.